system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -129,6 +171,12 @@ void PeriodicMeasurementNode::PerformPeriodicMeasurement()
 
 void PeriodicMeasurementNode::PublishStatisticMessage()
 {
+  if (!publisher_->is_activated()) {
+    RCLCPP_WARN(get_logger(), "PublishStatisticMessage: Lifecycle publisher is currently inactive."
+      " Messages will not be published.");
+    return;
+  }
```

Comment:
Could this happen? Why not throwing here? It seems like the node is broken if you end up in this situation.

Reply:
edit: actually in this world, the correct way to handle this is to trigger an ErrorProcessing transition IMHO
That seems to be the intent of the design: https://design.ros2.org/articles/node_lifecycle.html

It's the best of both world, the node won't die, but it also requires the user to restart the node so it can't become "ambient noise", which is the main problem with warnings.

Reply:
> Could this happen? Why not throwing here? It seems like the node is broken if you end up in this situation.

If our logic is broken then this could happen. When using a LifecyclePublisher, the activation / deactivation responsibility is on you (see the [talker demo](https://github.com/ros2/demos/blob/master/lifecycle/src/lifecycle_talker.cpp) for a nice example). Here we explicitly activate on Start() and deactivate on Stop().

With our tests for expected publisher state, it's worth replacing the log warning with an assert because it's a logic error.